### PR TITLE
Feature/backtrace function

### PR DESCRIPTION
**Pull request by deepweeb**
#34

# Introduction
Added the back trace method to return the name of the owner of the public key.
This is the solution to issue #34

# Purpose
When we display the contact list we need to get the name of each public key which is presented to us.

# Reason
When the function getBlocks(ownername) is used, all blocks will be returned with each public key in each block represent a user. We want to use this back trace function to get these users names in order to display the contacts by name of owner.

# Future Applicability
none

# PR Reflection
Not very difficult, a method is just added to a few classes and of course the testing.
Also, a little bit of refactoring took place, but they are very minor (see code changes).

### CODE REVIEW

File: app/src/test/java/nl/tudelft/b_b_w/controller/BlockControllerUnitTest.java [hunk 1]
```diff
@@ -71,6 +70,25 @@ public void testAddBlock() throws Exception {
...
+    /**
+     * getOwnerName test
+     * Test getting the owner name given public key.
+     */
+    @Test
+    public void getOwnerName() {
+        final String owner2 = "testowner";
+        final String publicKey2 = "testpubkey";
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, owner2, ownHash,
+                previousHashChain, previousHashSender, publicKey2, iban);
```
Comment (ashay123) [inline app/src/test/java/nl/tudelft/b_b_w/controller/BlockControllerUnitTest.java]: You missed the last parameter `trust`

File: app/src/test/java/nl/tudelft/b_b_w/model/BlockUnitTest.java [hunk 2]
```diff
@@ -36,7 +37,7 @@
      */
     @Before
     public void makeNewBlock() throws Exception {
-        _block = BlockFactory.getBlock(blockType, owner, ownHash,
+        _block = BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (ashay123) [inline app/src/test/java/nl/tudelft/b_b_w/model/BlockUnitTest.java]: Again, parameter `trust`.

File: app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java [hunk 3]
```diff
@@ -77,6 +77,17 @@ public BlockController(Context _context) {
         return res;
     }
 
+
+    /**
+     * Get the owner name, given the public key
+     * @param publicKey public key of the owner
+     *                  whose name we want to find
```
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java]: Why is this on a second line?

File: app/src/test/java/nl/tudelft/b_b_w/model/BlockUnitTest.java [hunk 4]
```diff
@@ -169,7 +170,7 @@ public void testTrustValueInit() {
      */
     @Test
     public void equalsTest() throws Exception {
-        final Block check = BlockFactory.getBlock(blockType, owner, ownHash,
+        final Block check = BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/BlockUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/BlockUnitTest.java [hunk 5]
```diff
@@ -183,7 +184,7 @@ public void equalsTest() throws Exception {
     @Test
     public void equalsFalseTest() throws Exception {
         final String _owner = "NOTOWNER";
-        final Block check = BlockFactory.getBlock(blockType, _owner, ownHash,
+        final Block check = BlockFactory.getBlock(TYPE_REVOKE, _owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/BlockUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 6]
```diff
@@ -53,7 +54,7 @@
     @Before
     public void setUp() {
         this.databaseHandler = new DatabaseHandler(RuntimeEnvironment.application);
-       _block =  BlockFactory.getBlock(blockType, owner, ownHash,
+       _block =  BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 7]
```diff
@@ -131,25 +132,44 @@ public void containsBlock2_false() {
     @Test
     public void getLatestSeqNum() {
 
-        final Block block2 = BlockFactory.getBlock(blockType, owner, ownHash,
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 8]
```diff
@@ -131,25 +132,44 @@ public void containsBlock2_false() {
...
+    /**
+     * getOwnerName test
+     * Test getting the owner name given public key.
+     */
+    @Test
+    public void getOwnerName() {
+        final String owner2 = "testowner";
+        final String publicKey2 = "testpubkey";
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, owner2, ownHash,
+                previousHashChain, previousHashSender, publicKey2, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 9]
```diff
@@ -131,25 +132,44 @@ public void containsBlock2_false() {
...
+
     /**
      * getLatestBlock test
      * Tests the latest block
      */
     @Test
     public void getLatestBlock() {
-        final Block block2 = BlockFactory.getBlock(blockType, "owner2", ownHash,
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, "owner2", ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 10]
```diff
@@ -131,25 +132,44 @@ public void containsBlock2_false() {
...
         databaseHandler.addBlock(_block);
         databaseHandler.addBlock(block2);
-        final Block expectBlock = BlockFactory.getBlock(blockType, "owner2", ownHash,
+        final Block expectBlock = BlockFactory.getBlock(TYPE_BLOCK, "owner2", ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 11]
```diff
@@ -160,11 +180,11 @@ public void getLatestBlock() {
      */
     @Test
     public void getBlockAfter() {
-        final Block block2 = BlockFactory.getBlock(blockType, owner, ownHash,
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 12]
```diff
@@ -160,11 +180,11 @@ public void getLatestBlock() {
...
         databaseHandler.addBlock(_block);
         databaseHandler.addBlock(block2);
-        final Block expectBlock = BlockFactory.getBlock(blockType, owner, ownHash,
+        final Block expectBlock = BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 13]
```diff
@@ -175,7 +195,7 @@ public void getBlockAfter() {
      */
     @Test
     public void getBlockBefore() {
-        final Block block2 = BlockFactory.getBlock(blockType, owner, ownHash,
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, owner, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

File: app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java [hunk 14]
```diff
@@ -189,7 +209,7 @@ public void getBlockBefore() {
     @Test
     public void getAllBlocks() {
         final String owner2 = "owner2";
-        final Block block2 = BlockFactory.getBlock(blockType, owner2, ownHash,
+        final Block block2 = BlockFactory.getBlock(TYPE_BLOCK, owner2, ownHash,
                 previousHashChain, previousHashSender, publicKey, iban);
```
Comment (humanjas) [inline app/src/test/java/nl/tudelft/b_b_w/model/DatabaseHandlerUnitTest.java]: trustvalue

**Comment by humanjas** [comment]
Code won't compile, if you don't fix those errors first.

**Comment by deepweeb** [comment]
No need for pull request anymore, my branch is fused with another branch to far and a power pull request that comes in the future will includes my changes.

**Pull request closed without merging**